server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1652,7 +1653,11 @@ private static void validateCompositeTemplate(
         final ClusterState stateWithIndex = ClusterState.builder(stateWithTemplate)
             .metadata(
                 Metadata.builder(stateWithTemplate.metadata())
-                    .put(IndexMetadata.builder(temporaryIndexName).settings(finalResolvedSettings))
+                    .put(
+                        IndexMetadata.builder(temporaryIndexName)
+                            .eventIngestedRange(IndexLongFieldRange.UNKNOWN) // necessary to pass asserts in ClusterState constructor
```

Comment:
I would prefer to avoid the package private call here (while technically in the same package it is very much client code of the `IndexMetadata` class). Can we instead do:

```suggestion
                            .eventIngestedRange(IndexLongFieldRange.NO_SHARDS, state.getMinTransportVersion()) // necessary to pass asserts in ClusterState constructor
```

??